packages/app/src/providers/api.ts

```diff
@@ -93,21 +74,16 @@ export abstract class DataProviderApi {
         );
       }
       throw error;
-    } finally {
-      // Remove progress when request fulfills
-      this.progress.delete(storeParams);
-      this.notifyProgressChange();
     }
   }
 
-  private notifyProgressChange() {
-    this.progressListeners.forEach((cb) => cb([...this.progress.values()]));
-  }
-
   public abstract getEntity(path: string): Promise<ProvidedEntity>;
+
   public abstract getValue(
     params: ValuesStoreParams,
     signal?: AbortSignal,
+    onProgress?: OnProgress,
```

Comment:
Both `signal` and `onProgress` are optional so I don't have to pass them when testing the provider APIs.